### ISSUE TITLE
`get_path_prefix()` broken on Windows

### ISSUE THREAD

**Issue by mih** [datalad/datalad#2562]
```
[00:15:30] ======================================================================
[00:15:30] FAIL: datalad.tests.test_utils.test_path_prefix
[00:15:30] ----------------------------------------------------------------------
[00:15:30] Traceback (most recent call last):
[00:15:30]   File "C:\Miniconda35\envs\test-environment\lib\site-packages\nose\case.py", line 198, in runTest
[00:15:30]     self.test(*self.arg)
[00:15:30]   File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\tests\utils.py", line 669, in newfunc
[00:15:30]     return t(*(arg + (filename,)), **kw)
[00:15:30]   File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\tests\test_utils.py", line 696, in test_path_prefix
[00:15:30]     eq_(get_path_prefix('/d1/d2', '/d1/d2/d3'), '..')
[00:15:30] AssertionError: '/d1/d2' != '..'

```

**Comment by mih** [comment]
https://github.com/datalad/datalad-revolution uses pathlib to avoid this code entirely

**Comment by mih** [comment]
The release of 0.12 with this fix is imminent (#3993) closing.
